src/types/winston.ts

```diff
@@ -0,0 +1,53 @@
+import { BigNumber } from 'bignumber.js';
+
+export class Winston {
+	private amount: BigNumber;
+	constructor(amount: BigNumber.Value) {
+		this.amount = new BigNumber(amount);
+		if (this.amount.isLessThan(0) || !this.amount.isInteger()) {
+			throw new Error('Winston value should be a non-negative integer!');
+		}
+	}
+
+	plus(winston: Winston): Winston {
+		return W(this.amount.plus(winston.amount));
+	}
+
+	minus(winston: Winston): Winston {
+		return W(this.amount.minus(winston.amount));
+	}
+
+	times(multiplier: BigNumber.Value): Winston {
+		return W(this.amount.times(multiplier).decimalPlaces(0, BigNumber.ROUND_DOWN));
+	}
+
+	dividedBy(divisor: BigNumber.Value): Winston {
+		// TODO: Best rounding strategy? Up or down?
+		return W(this.amount.dividedBy(divisor).decimalPlaces(0, BigNumber.ROUND_CEIL));
+	}
+
+	isGreaterThan(winston: Winston): boolean {
+		return this.amount.isGreaterThan(winston.amount);
+	}
+
+	static difference(a: Winston, b: Winston): string {
+		return a.amount.minus(b.amount).toString();
+	}
+
+	toString(): string {
+		return this.amount.toFixed();
+	}
+
+	valueOf(): string {
+		return this.amount.toFixed();
+	}
+
+	static max(...winstons: Winston[]): Winston {
+		BigNumber.max();
+		return winstons.reduce((max, next) => (next.amount.isGreaterThan(max.amount) ? next : max));
+	}
+}
```

Comment:
We can maybe use `equals()` for comparison (?)
```suggestion

	static equals(a: Winston, b: Winston): boolean {
		return a.amount.equals(b.amount);
	}
}
```

Reply:
How does this help in the context of Max?

Reply:
We don't seem to do any Winston equality checking in the code other than in the tests, and in that context we don't have the agency to call equals directly without eliminating some of the test expectation output beautification we get from chai.